Server.Core/RequestBuilder.cs

```diff
@@ -0,0 +1,122 @@
+﻿ using System;
+ using System.Text;
+
+namespace Server.Core
+{
+    public class RequestBuilder
+    {
+        private Request request;
+
+        public RequestBuilder()
+        {
+            request = new Request();
+        }
+
+        public Request BuildRequest()
+        {
+            return request;
+        }
+
+        public Request BuildRequestFromWholeMessage(byte[] requestMessage)
+        {
+            if (requestMessage.Length == 0)
+            {
+                SetRequestToNoValues();
+            }
+            else
+            {
+                ParseMessageBytes(requestMessage);
+            }
+            return request;
+        }
+
+        public void SetRequestHttpVersion(string httpVersion)
+        {
+            request.HttpVersion = httpVersion;
+        }
+
+        public void SetRequestUri(string uri)
+        {
+            request.Uri = uri;
+        }
+
+        public void SetRequestMethod(string method)
+        {
+            request.Method = method;
+        }
+
+        public void SetRequestHeaders(string[] headers)
+        {
+            request.Headers = headers;
+        }
+
+        public void SetRequestBody(byte[] body)
+        {
+            request.Body = body;
+        }
+
+        public void SetRequestBody(string bodyMessage)
+        {
+            request.Body = Encoding.UTF8.GetBytes(bodyMessage);
+        }
+
+        private byte[] GetBodyOfMessage(int startIndex, byte[] message)
+        {
+            var body = new byte[message.Length - startIndex];
+            for (var index = startIndex; index < message.Length; index++)
+            {
+                body[index-startIndex] = message[index];
+            }
+            return body;
+        }
+
+        private string[] DivideHeaders(string[] allLines)
```

Comment:
I'm a little confused about your second point.  What do you mean by the parsing needn't be public to the world? Everything that uses this API doesn't know about the way it creates the requests and all of the parsing methods are private.